src/main/java/duke/main/Ui.java

```diff
@@ -0,0 +1,200 @@
+package duke.main;
+
+import java.util.ArrayList;
+
+import duke.gui.MainWindow;
+import duke.task.Task;
+
+/**
+ * A class to handle user input and output for the user.
+ */
+public class Ui {
+    private static final String[] GREETING = {"Hi, I'm Duke.\nWhat can I do for you?\nI'll do my best! :)"};
+    private static final String[] BYE = {"NOOOOOO... Don't send me back to the void! T_T"};
+
+    private static final String[] BEFORE_LIST = {"Here's your task list!"};
+    private static final String[] AFTER_LIST = {"I'm useful right?"};
+    private static final String[] ADD_LIST = {"Yep! I've added this task: "};
+    private static final String[] DELETE_LIST = {"OK! I've removed this task: "};
+
+    private static final String[] MARK_DONE = {"I've marked this task as done:"};
+    private static final String[] MARK_UNDONE = {"I've marked this task as not done:"};
+    private static final String[] PREV_DONE = {"Just to let you know this task was already done previously."};
+    private static final String[] PREV_UNDONE = {"Just to let you know this task was already not done previously."};
+
+    private static final String[] MISSING_TIME = {"I think you need a time for this."};
+    private static final String[] MISSING_INDEX = {"Please provide an index in the command."};
+    private static final String[] MISSING_DESCRIPTION = {"I need a description for the task..."};
+    private static final String[] MISSING_ARGUMENT = {"I need an argument for this command..."};
+    private static final String[] NOT_A_NUMBER = {"I don't think that's a number..."};
+    private static final String[] CANNOT_SAVE = {"I can't seem to save the task list..."};
+
+    private static final String[] INVALID_COMMAND = {"I don't quite get what you're saying..."};
+    private static final String[] INVALID_INDEX = {"Umm... I think this task number does not exist."};
+    private static final String[] INVALID_TIME = {"I'm sorry but I don't understand your time format."};
+
+    private int messageStatus = 0;
+    private MainWindow mainWindow;
+
+    /**
+     * Sets the MainWindow object for Ui to use.
+     * @param window
+     */
+    public void setMainWindow(MainWindow window) {
+        this.mainWindow = window;
+    }
+
+    private void outputMessage(String message) {
+        mainWindow.addDukeDialog(message);
+    }
+
+    /**
+     * Generates a message for showing total number of Task objects.
+     * @param num Number to display.
+     * @return The message for showing total number of Task objects.
+     */
+    private String getListSizeMsg(int num) {
+        String[] listCount = {"I'm keeping track of " + num + " task(s) currently!"};
+        return listCount[this.messageStatus];
+    }
+
+    /**
+     * Shows the greeting message.
+     */
+    public void showGreeting() {
+        this.outputMessage(GREETING[this.messageStatus]);
+    }
```

Comment:
I like the way you named all the functions in this class. It is very consistent and clear.